mesonbuild/interpreter/interpreter.py

```diff
@@ -2310,8 +2310,6 @@ def func_install_emptydir(self, node: mparser.BaseNode, args: T.Tuple[str], kwar
         d = build.EmptyDir(args[0], kwargs['install_mode'], self.subproject, kwargs['install_tag'])
         self.build.emptydir.append(d)
 
-        return d
-
```

Comment:
I am not sure I understand what you mean… The annotation implies the return value is unused, so why have the excess code?

> Every other function here returns the `build.*` object it creates, and is annotated to do so.

That's not true, `func_subdir`, `func_add_test_setup`, etc return nothing.

Reply:
Sorry, I meant the other `install_*()` functions.

The annotation does NOT imply the return value is used, the annotation implies that the return value is `None` but that isn't actually true.

Reply:
Ah, okay. I'm unclear though, if `None` return type is incorrect, which one is then? I presume that after this PR it ought to be `List[build.EmptyDir]`, but this differs from other `install_*()` functions, and… like, nothing falls out due to changing the type: tests do not break, linter does not complain, Meson works as usual…

That's actually another reason why I believe that `None` is the correct type, because clearly nothing uses the return value of the function, so why have it return anything?

Reply:
Should I perhaps make other `install_*()` functions change to also return `None`…? I might be wrong as I did not test that, but offhand I would guess that nothing might be using their return values as well.